text/0000-edition-2021-and-beyond.md

```diff
@@ -0,0 +1,207 @@
+- Feature Name: N/A
+- Start Date: 2020-07-29
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+
+[summary]: #summary
+
+- Announce plans for a Rust 2021 Edition, and for a regular cadence of editions every 3 years thereafter.
+  - We will roll out an edition regardless of whether there are breaking changes.
+- Unlike Rust 2018, we will avoid using editions as a "deadline" to tie together high-priority projects.
+  - Instead, we embrace the train model, but editions are effectively a "somewhat bigger release", giving us an opportunity to give an overview of all the work that has landed over the previous three years.
+- We specify a cadence for Edition lints.
+  - "Edition idiom" lints for Edition N will warn for editions before N, and become "deny by default" in Edition N.
+  - Since it would be disruptive to introduce deny-by-default lints for Rust 2018 now, the Rust 2018 lints are repurposed into Rust 2021 Edition lints.
+- We specify a policy on reserving keywords and other prospective changes.
+  - In short, reserving keywords is allowed only as part of an active project group.
+
+# Motivation
+
+[motivation]: #motivation
+
+The plan for editions was laid out in [RFC 2052] and Rust had its first edition in 2018. This effort was in many ways a success but also resulted in some difficult lessons. As part of this year's roadmap, one of the major questions we identified was that we need to decide whether we are going to do more editions and -- if so -- how we are going to manage the process.
+
+[rfc 2052]: https://github.com/rust-lang/rfcs/blob/master/text/2052-epochs.md
+
+This RFC proposes various clarifications to the edition process going forward:
+
+- We will do new Rust editions on a regular, three-year cadence.
+  - We will roll out an edition regardless of whether there are breaking changes.
+- Unlike Rust 2018, we will avoid using editions as a "deadline" to tie together high-priority projects.
+  - Instead, we embrace the train model, but editions are effectively a "somewhat bigger release", giving us an opportunity to give an overview of all the work that has landed over the previous three years.
+- We specify a cadence for Edition lints.
+  - "Edition idiom" lints for Edition N will warn for editions before N, and become "deny by default" in Edition N.
```

Comment:
i could imagine a universe where idiom edition 20XX and idiom 20XX+6 conflict with each other. avoiding this is something i could imagine everyone is motivated by, but not 100% possible. tooling wise this could be really confusing, as we prepare the lints we should probably add a step to audit for this and consider retiring old lints?